pkg/controller/kibana/pod.go

```diff
@@ -115,6 +129,51 @@ func GetKibanaContainer(podSpec corev1.PodSpec) *corev1.Container {
 	return pod.ContainerByName(podSpec, kbv1.KibanaContainerName)
 }
 
+func GetKibanaBasePathFromSpecEnv(podSpec corev1.PodSpec) string {
+	kbContainer := GetKibanaContainer(podSpec)
+	if kbContainer == nil {
+		return ""
+	}
+	for _, envVar := range kbContainer.Env {
+		if envVar.Name == "SERVER_BASEPATH" {
+			return envVar.Value
+		}
+	}
+	return ""
+}
+
 func getDefaultContainerPorts(kb kbv1.Kibana) []corev1.ContainerPort {
 	return []corev1.ContainerPort{{Name: kb.Spec.HTTP.Protocol(), ContainerPort: int32(network.HTTPPort), Protocol: corev1.ProtocolTCP}}
 }
+
+func getKibanaBasePath(kb kbv1.Kibana) (string, error) {
+	if kbBasePath := GetKibanaBasePathFromSpecEnv(kb.Spec.PodTemplate.Spec); kbBasePath != "" {
+		return kbBasePath, nil
+	}
+
+	if kb.Spec.Config == nil {
+		return "", nil
+	}
+	kbucfgConfig, err := ucfg.NewFrom(kb.Spec.Config.Data)
+	if err != nil {
+		return "", err
+	}
+
+	kbCfg := kibanaConfig{}
+	err = kbucfgConfig.Unpack(&kbCfg)
+	if err != nil {
+		return "", err
+	}
+
+	if kbCfg.Server.BasePath != "" {
+		// We give preference to base path set in the spec
+		return kbCfg.Server.BasePath, nil
+	}
+
+	// Check for a flattened structure
+	if kbucfgConfig.HasField("server.basePath") {
+		return kbucfgConfig.String("server.basePath", -1)
+	}
```

Comment:
This is unnecessary if you either re-use `CanoncialConfig` or alternatively pass the `PathSep(".")` option

```golang
var Options = []ucfg.Option{ucfg.PathSep("."), ucfg.AppendValues}
```
when parsing the config.

Reply:
Thanks, I made this change.